code/datums/crew_objective.dm

```diff
@@ -141,36 +156,49 @@ ABSTRACT_TYPE(/datum/objective/crew/chiefengineer)
 /datum/objective/crew/chiefengineer/furnaces
 	explanation_text = "Make sure all furnaces on the station are active at the end of the round."
 	medal_name = "Slow Burn"
+	var/static/check_result = null
 	check_completion()
-		for(var/obj/machinery/power/furnace/F in machine_registry[MACHINES_POWER])
-			if(F.z == Z_LEVEL_STATION && !F.active && istype(get_area(F), /area/station))
-				return FALSE
-		return TRUE
+		if(isnull(check_result))
+			check_result = TRUE
+			for(var/obj/machinery/power/furnace/F in machine_registry[MACHINES_POWER])
+				if(F.z == 1 && !F.active && istype(F.loc.loc, /area/station))
```

Comment:
```suggestion
				if(F.z == Z_LEVEL_STATION && !F.active && istype(F.loc.loc, /area/station))
```
why are you changing this fix for others

Reply:
It seems like that might have been from when I merged from master after #6244 was merged; I'll double check to make sure no other mistakes arose from the merge, though that'll be tomorrow since I'm going to sleep soon